### PR TITLE
[WIP] support get flops and parameters in dev-1.x

### PR DESCRIPTION
**Pull request by vansin**
## Motivation

mmocr 1.x doesn't support the get flops

## Modification

add the get_flops script and the doc

### CODE REVIEW

File: docs/zh_cn/user_guides/useful_tools.md [hunk 1]
```diff
@@ -50,3 +50,69 @@ python tools/analysis_tools/offline_eval.py configs/textdet/psenet/psenet_r50_fp
...
+|    det_head.binarize.4       |    0.128K              |    83.886M |
+|    det_head.binarize.6       |    0.257K              |    67.109M |
+|   det_head.threshold         |   0.164M               |            |
+|    det_head.threshold.0      |    0.147M              |            |
+|    det_head.threshold.1      |    0.128K              |            |
+|    det_head.threshold.3      |    16.448K             |            |
+|    det_head.threshold.4      |    0.128K              |            |
+|    det_head.threshold.6      |    0.257K              |            |
+!!!Please be cautious if you use the results in papers. You may need to check if all ops are supported and verify that the flops computation is correct.
+```
```
Comment (xinke-wang) [inline docs/zh_cn/user_guides/useful_tools.md:118]: Add an arg table like other scripts.

| 参数          | 类型  | 说明                                     |
| ------------- | ----- | ---------------------------------------- |
| config        |    |                   |
| shape        |   |                 |
| size-divisor   |   |      |
| --cfg-options |  ||

Also we can fix the --cfg-options example link in this markdown to `user_guides/config.html#命令行修改配置`

Comment (vansin) [reply docs/zh_cn/user_guides/useful_tools.md:118]: ok

**Review by gaotongxiao — COMMENTED** [review]
We can provide English docs as well.

File: docs/zh_cn/user_guides/useful_tools.md [hunk 2]
```diff
@@ -50,3 +50,80 @@ python tools/analysis_tools/offline_eval.py configs/textdet/psenet/psenet_r50_fp
 | config        | str   | （必须）配置文件路径。                   |
 | pkl_results   | str   | （必须）预先保存的预测结果文件。         |
 | --cfg-options | float | 用于覆写配置文件中的指定参数。[示例](<>) |
+
+### 计算FLOPs和参数量
```
Comment (Harold-lkk) [inline docs/zh_cn/user_guides/useful_tools.md:54]:
```suggestion
### 计算 FLOPs 和参数量
```

File: tools/analysis_tools/get_flops.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    elif len(args.shape) == 2:
+        h, w = args.shape
+    else:
+        raise ValueError('invalid input shape, please use --shape h w')
+
+    input_shape = (1, 3, h, w)
+
+    print('input shape is ', input_shape)
+
+    # model = init_detector(args.config, device='cpu')  # or device='cuda:0'
```
Comment (Harold-lkk) [inline tools/analysis_tools/get_flops.py:42]: remove

File: docs/zh_cn/user_guides/useful_tools.md [hunk 4]
```diff
@@ -50,3 +50,80 @@ python tools/analysis_tools/offline_eval.py configs/textdet/psenet/psenet_r50_fp
...
+我们提供一个计算 FLOPs 和参数量的方法，首先我们使用以下命令安装依赖。
+
+```shell
+pip install fvcore
+```
+
+计算 FLOPs 和参数量的脚本使用方法如下：
+
+```shell
+python tools/analysis_tools/get_flops.py ${config} --shape ${IMAMGE_SHAPE}
```
Comment (gaotongxiao) [inline docs/zh_cn/user_guides/useful_tools.md:65]:
```suggestion
python tools/analysis_tools/get_flops.py ${config} --shape ${IMAGE_SHAPE}
```

File: docs/zh_cn/user_guides/useful_tools.md [hunk 5]
```diff
@@ -50,3 +50,80 @@ python tools/analysis_tools/offline_eval.py configs/textdet/psenet/psenet_r50_fp
...
+```
+
+| 参数    | 类型 | 说明                                                                 |
+| ------- | ---- | -------------------------------------------------------------------- |
+| config  | str  | （必须) 配置文件路径。                                               |
+| --shape | int  | 计算 FLOPs 使用的图片尺寸，实例用法 `--shape 320 320` 默认为 640 640 |
```
Comment (gaotongxiao) [inline docs/zh_cn/user_guides/useful_tools.md:71]:
```suggestion
| --shape | int*2 | 计算 FLOPs 使用的图片尺寸，如 `--shape 320 320`。 默认为 `640 640` |
```

File: docs/en/user_guides/useful_tools.md [hunk 6]
```diff
@@ -50,3 +50,80 @@ In addition, based on this tool, users can also convert predictions obtained fro
 | config        | str   | (required) Path to the config.    |
 | pkl_results   | str   | (required) The saved predictions. |
 | --cfg-options | float | Override configs. [Example](<>)   |
+
+### Calculate FLOPs and Parameters
+
+We provide a method to calculate the FLOPs and Parameters, first we install the dependencies using the following command.
```
Comment (gaotongxiao) [inline docs/en/user_guides/useful_tools.md:56]:
```suggestion
We provide a method to calculate the FLOPs and the number of parameters, first we install the dependencies using the following command.
```

File: docs/en/user_guides/useful_tools.md [hunk 7]
```diff
@@ -50,3 +50,80 @@ In addition, based on this tool, users can also convert predictions obtained fro
 | config        | str   | (required) Path to the config.    |
 | pkl_results   | str   | (required) The saved predictions. |
 | --cfg-options | float | Override configs. [Example](<>)   |
+
+### Calculate FLOPs and Parameters
```
Comment (gaotongxiao) [inline docs/en/user_guides/useful_tools.md:54]:
```suggestion
### Calculate FLOPs and the Number of Parameters
```

File: docs/en/user_guides/useful_tools.md [hunk 8]
```diff
@@ -50,3 +50,80 @@ In addition, based on this tool, users can also convert predictions obtained fro
...
+
+We provide a method to calculate the FLOPs and Parameters, first we install the dependencies using the following command.
+
+```shell
+pip install fvcore
+```
+
+The script to calculate the FLOPs and Parameters is used as follows.
```
Comment (gaotongxiao) [inline docs/en/user_guides/useful_tools.md:62]:
```suggestion
The usage of the script to calculate FLOPs and the number of parameters is as follows.
```

File: docs/en/user_guides/useful_tools.md [hunk 9]
```diff
@@ -50,3 +50,80 @@ In addition, based on this tool, users can also convert predictions obtained fro
...
+```shell
+python tools/analysis_tools/get_flops.py ${config} --shape ${IMAMGE_SHAPE}
+```
+
+| ARGS    | Type | Description                                                                                  |
+| ------- | ---- | -------------------------------------------------------------------------------------------- |
+| config  | str  | (required) Path to the config.                                                               |
+| --shape | int  | Image size to use when calculating FLOPs, example usage `--shape 320 320` Default is 640 640 |
+
+The sample command to get `dbnet_resnet18_fpnc_100k_synthtext.py` FLOPs and the number of parameters is as follows.
```
Comment (gaotongxiao) [inline docs/en/user_guides/useful_tools.md:73]:
```suggestion
For example, you can run the following command to get FLOPs and the number of parameters of `dbnet_resnet18_fpnc_100k_synthtext.py`:
```

File: docs/zh_cn/user_guides/useful_tools.md (hunk 4 above)
Comment (gaotongxiao) [reply docs/zh_cn/user_guides/useful_tools.md:65]: fix eng version accordingly

File: docs/zh_cn/user_guides/useful_tools.md (hunk 5 above)
Comment (gaotongxiao) [reply docs/zh_cn/user_guides/useful_tools.md:71]: fix eng version accordingly

**Review by gaotongxiao — APPROVED** [review]

**Pull request merged**